src/components/button.js

```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+import Link from 'next/link';
+import PropTypes from 'prop-types';
+
+export default class Button extends Component {
```

Comment:
There's no needs for "Class" Components for simple cases (with no state manipulation) as this button.

You should use Stateless Component in this case (https://hackernoon.com/react-stateless-functional-components-nine-wins-you-might-have-overlooked-997b0d933dbc). It's more concise and clearer.

```
const Button = ({ color, icon, label, path }) => (
    <Link href={{ pathname: path }}>
        <a className={`btn ${color ? color : ''}`}>
            <i className="material-icons left">{icon}</i>
            {label}
        </a>
    </Link>
);

Button.propTypes = {...};

export default Button;
```

Reply:
You're component don't need to re-render each time props changes. If props are the same, there's no reason to re-render. It's called "purity", with "pure" components.

With "class" component, you should extends 
[PureComponent](https://reactjs.org/docs/react-api.html#reactpurecomponent).

With stateless component, you should "pure" HOC from recompose https://github.com/acdlite/recompose/blob/master/docs/API.md#pure

Reply:
I advice you to read the article below. It'll help you to understand performance considerations in React.

https://marmelab.com/blog/2017/02/06/react-is-slow-react-is-fast.html